http-server-netty/src/main/java/io/micronaut/http/server/netty/body/ImmediateByteBody.java

```diff
@@ -45,12 +48,14 @@
  * @author Jonas Konrad
  */
 @Internal
-public final class ImmediateByteBody extends ManagedBody<ByteBuf> implements ByteBody {
+public final class ImmediateByteBody extends ManagedBody<ByteBuf> implements ByteBody, ByteBuffer<ByteBuf> {
```

Comment:
just access the ByteBuf value directly instead and wrap it in NettyByteBuffer, no need to delegate this much

Reply:
then I would need to couple the filter logic to netty which I can't do the way the current code is structured

Reply:
i also think that conversion to ByteBuffer should fail if the body is already claimed, it should not be an issue in filters and will prevent race conditions

Reply:
not sure what you mean here, maybe you want to send a PR with these improvements

Reply:
sure

Reply:
i think i can make it work without FULL_CONTENT too, i will do it after lunch.